crates/polars-time/src/group_by/dynamic.rs

```diff
@@ -311,16 +313,109 @@ impl Wrap<&DataFrame> {
         let groups = if by.is_empty() {
             let vals = dt.downcast_iter().next().unwrap();
             let ts = vals.values().as_slice();
-            let (groups, lower, upper) = group_by_windows(
-                w,
-                ts,
-                options.closed_window,
-                tu,
-                tz,
-                include_lower_bound,
-                include_upper_bound,
-                options.start_by,
-            );
+
+            let vanilla_start_step = (ts[0] == 0) && (ts[1] - ts[0] == 1);
```

Comment:
Can we move this to a separate function so that implementation is separate from dispatch?

Reply:
Moved the logic to a separate function in the group_by.rs file